src/sgx/signing/signing_trusted.c

```diff
@@ -0,0 +1,53 @@
+/*
+ * This file contains Solana's SGX enclave code for signing data.
+ */
+
+#include <stdbool.h>
+#include <string.h>
+
+#include "ed25519.h"
+#include "signing_t.h"
+
+static bool initialized;
+static uint8_t public_key[32], private_key[64];
+
+/* This function creates a new public/private keypair in
+   enclave trusted space.
+*/
+sgx_status_t init(uint32_t keylen, uint8_t* pubkey) {
+  if (keylen < sizeof(public_key)) {
+    return SGX_ERROR_INVALID_PARAMETER;
+  }
+
+  uint8_t seed[32];
+  sgx_status_t status = sgx_read_rand(seed, sizeof(seed));
+  if (SGX_SUCCESS != status) {
+    return status;
+  }
+
+  ed25519_create_keypair(public_key, private_key, seed);
+  memcpy(pubkey, public_key, sizeof(public_key));
+
+  initialized = true;
+
+  return SGX_SUCCESS;
+}
+
+/* This function signs the msg using private key.
+ */
+sgx_status_t sign(uint32_t msg_len,
+                  uint8_t* msg,
+                  uint32_t sig_len,
+                  uint8_t* signature) {
+  if (!initialized) {
+    return SGX_ERROR_INVALID_STATE;
+  }
+
+  if (sig_len < 64) {
```

Comment:
Can we move this magic 64 up to a const or use a `sizeof()` here?

Reply:
We'll probably have to use a const. It's there because ed25519 library expects these sizes, but does not have a typedef (neither it checks or returns an error).